docs/advanced_usage.md

```diff
@@ -418,6 +423,69 @@ Insurances/                             # Insurances
     Defining a storage path is optional. If no storage path is defined for a
     document, the global [`PAPERLESS_FILENAME_FORMAT`](configuration.md#PAPERLESS_FILENAME_FORMAT) is applied.
 
+### Filename Templates {#filename-templates}
+
+The filename formatting using the [Django Template Language](https://docs.djangoproject.com/en/5.1/ref/templates/language/) to build the filename.
+This allows for complex logic to be included in the format, including [tags](https://docs.djangoproject.com/en/5.1/ref/templates/language/#tags)
+and [filters](https://docs.djangoproject.com/en/5.1/ref/templates/language/#filters) to manipulate the [variables](#filename-format-variables)
+provided.
+
+In addition, the entire Document instance is available to be utilized in a more advanced way, as well as some variables which only make sense to be accessed
+with more complex logic.
+
+#### Additional Variables
+
+- `{{ tag_name_list }}`: A list of tag names applied to the document, ordered by the tag name. Note this is a list, not a single string
+- `{{ custom_fields }}`: A mapping of custom field names to their type and value. A user can access the mapping by field name or check if a field is applied by checking its existence in the variable.
+
+#### Examples
+
+This example will construct a path based on the archive serial number range:
+
+```django
+somepath/
```

Comment:
Should we raise the character limit? 

https://github.com/paperless-ngx/paperless-ngx/blob/8fd34f6b195211a1668b753667200bbd96f114e7/src/documents/models.py#L127

This example can't actually be set:
<img width="362" alt="Screenshot 2024-10-02 at 9 24 09 PM" src="https://github.com/user-attachments/assets/28c400c3-d2fc-4a96-9f77-aac31a5eb066">

Reply:
> Should we raise the character limit?

Be aware that some OS have limits for their storage paths char length. In Windows it's 260 character max path length before Win10 allows users to removed it by editing their registry.
Unix: max filename length: 255 bytes or 255chars depending on the source online.

Reply:
Yes, the new migration will increase this. I'm thinking at least doubling it?  That seems enough?

Reply:
Well it doesn't help if you double it to 512, if the underlying OS doesn't support it. Or what exactly is your plan?

Reply:
That is already handled elsewhere, we’re referring to the template field in the database friend

Reply:
I decided to 4x it.  Might as well make it larger all at once to fix the no doubt complicated templates which will be dreamed up